### PR TITLE
New Type for Variants (Also for sizes now)

### PR DESCRIPTION
**Pull request by Knerio**
This would resolve https://github.com/code0-tech/pictor/issues/50

With https://github.com/code0-tech/pictor/pull/66/commits/d61bda0deddb1d0997a6b4992778f984562f08a6 there are also sizes like xs, md and xl etc

### CODE REVIEW

**Review by nicosammito — DISMISSED** [review]
Overall nicely done, but as you see in #59 we are changing from Variant to Color, because Variant is used differently now

**Comment by Knerio** [comment]
> Overall nicely done, but as you see in #59 we are changing from Variant to Color, because Variant is used differently now

Thats true, but in https://github.com/code0-tech/pictor/pull/64 that would be a problem

**Comment by Knerio** [comment]
> > Overall nicely done, but as you see in #59 we are changing from Variant to Color, because Variant is used differently now
> 
> Thats true, but in #64 that would be a problem

Or we could implement these styles but we also need to fix the @include box then for all the new variants

**Comment by nicosammito** [comment]
> > Overall nicely done, but as you see in #59 we are changing from Variant to Color, because Variant is used differently now
> 
> Thats true, but in #64 that would be a problem

Yeh we need to change this in all components from now on. Just change variant: “bla” | “bla” to color: “bla” | “bla”

File: src/components/alert/Alert.tsx [hunk 1]
```diff
@@ -50,12 +51,12 @@ export interface AlertHeadingType {
 }
 
 export interface AlertIconType {
-    variant: "primary" | "secondary" | "info" | "success" | "warning" | "error"
+    variant: Color
```
Comment (nicosammito) [inline src/components/alert/Alert.tsx:54]: should be renamed also to color

File: src/styles/_variables.scss [hunk 2]
```diff
@@ -8,11 +9,25 @@ $error: #D90429;
...
+$lg: ("font": 1.3rem);
+$xl: ("font": 2rem);
+
+$sizes: (
+        "xs": $xs,
+        "sm": $sm,
+        "md": $md,
+        "lg": $lg,
+        "xl": $xl,
+);
```
Comment (nicosammito) [inline src/styles/_variables.scss:26]: parse in directly the font sizes rather to map in map

File: src/components/button/Button.stories.tsx [hunk 3]
```diff
@@ -36,9 +37,8 @@ export const Buttons: ButtonStory = {
 
         return <>
             {
-                ["primary", "secondary", "info", "success", "warning", "error"].map(value => {
-                    // @ts-ignore
-                    return <Button disabled={disabled} variant={variant} color={value}>
+                Colors.map(value => {
+                    return <Button disabled={disabled} color={value}>
```
Comment (nicosammito) [inline src/components/button/Button.stories.tsx:41]: why you are removing the variant. Button have a variant and a color

File: src/components/FontSizes/Font.style.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
+@import "../../styles/variables";
+
+
+@each $name, $entry in $sizes {
+  .size__#{$name} {
```
Comment (nicosammito) [inline src/components/FontSizes/Font.style.scss:5]: look at BEM. should be size--

File: src/components/FontSizes/Font.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import {Size} from "../../utils/utils";
+import "./Font.style.scss"
+
+export interface FontType {
+    children: string,
+    size: Size,
+}
+
+const Font: React.FC<FontType> = (props) => {
```
Comment (nicosammito) [inline src/components/FontSizes/Font.tsx:10]: rename font to Text

File: src/components/FontSizes/Font.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    const {size, children} = props;
+
+    return <text className={`size__${size}`}>
```
Comment (nicosammito) [inline src/components/FontSizes/Font.tsx:13]: should be a span

File: src/components/FontSizes/Font.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import {Size} from "../../utils/utils";
+import "./Font.style.scss"
+
+export interface FontType {
```
Comment (nicosammito) [inline src/components/FontSizes/Font.tsx:5]: allow also all html span props

File: src/utils/utils.ts [hunk 8]
```diff
@@ -1,5 +1,14 @@
...
 
+export type Color = "primary" | "secondary" | "info" | "success" | "warning" | "error";
+
+export const Colors: Color[] = ["primary", "secondary", "info", "success", "warning", "error"]
+
+export type Size = "xs" | "sm" | "md" | "lg" | "xl";
+
+export const Sizes: Size[] = ["xs", "sm", "md", "lg", "xl"]
+
+
```
Comment (nicosammito) [inline src/utils/utils.ts:11]: outsource this to a Color file inside the util folder

**Review by nicosammito — CHANGES_REQUESTED** [review]

**Review by nicosammito — CHANGES_REQUESTED** [review]
Just some minor changes before we can merge

File: src/components/FontSizes/Text.stories.tsx [hunk 9]
```diff
@@ -1,24 +1,34 @@
...
-                )
-            }
-        </ButtonGroup>
+
+export const TextSizes: StoryObj<typeof Text> = {
+    render: (args) => {
+        const {size} = args
+
+        return <>
+            <Button>
```
Comment (nicosammito) [inline src/components/FontSizes/Text.stories.tsx:26]: Could you also remove the button. Let’s just show the text

File: src/components/FontSizes/Text.tsx [hunk 10]
```diff
@@ -1,8 +1,8 @@
-import React from "react";
+import React, {HTMLProps} from "react";
 import {Size} from "../../utils/types";
 import "./Text.style.scss"
 
-export interface FontType extends Omit<Omit<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, "children">, "className"> {
+export interface FontType extends Omit<Omit<Omit<HTMLProps<HTMLSpanElement>, "children">, "className">, "size"> {
```
Comment (nicosammito) [inline src/components/FontSizes/Text.tsx:5]: Do we need to omit the className?

**Review by nicosammito — CHANGES_REQUESTED** [review]

File: src/components/FontSizes/Text.stories.tsx (hunk 9 above)
Comment (Knerio) [reply src/components/FontSizes/Text.stories.tsx]: [65e7129](https://github.com/code0-tech/pictor/pull/66/commits/65e7129408c22c5d3580c440ef2e00b7df5101dc)

File: src/components/FontSizes/Text.tsx (hunk 10 above)
Comment (Knerio) [reply src/components/FontSizes/Text.tsx]: [2c063cd](https://github.com/code0-tech/pictor/pull/66/commits/2c063cdac914350be58057316b5a2e55208569ec)

**Review by nicosammito — APPROVED** [review]
Nicely done

**Review by nicosammito — APPROVED** [review]

**Pull request merged**